### DISCUSSION TITLE
RoleArn for AWS::Serverless::StateMachine -> Events -> CWSchedule  for Type: Schedule

### DISCUSSION THREAD

**Discussion by roypronoy** [aws/serverless-application-model#3290] [Ideas]
### Description:

In the Resource [AWS::Serverless::StateMachine](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-statemachine.htmlhttps://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-statemachine.html), the User should be able to use their own Role under [Events](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-statemachine.html#sam-statemachine-events) -> [CWSchedule](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-property-statemachine-statemachineeventsource.html)  for [Type: Schedule](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-property-statemachine-statemachineschedule.html) 

Currently, by default, a new Role is created under the Logical Resource id `StateMachineCWScheduleRole` and on expansion, this Role is associated with the dynamically created `AWS::Events::Rule` Resource. 

Sample:

```
Resources:
  ServerlessStateMachine:
    Type: AWS::Serverless::StateMachine
    Properties:
      Type: STANDARD
      Name: SampleStateMachine
      DefinitionUri:
        Bucket: mybucket
        Key: StateMachineDefinition.json
      Role: arn:aws:iam::XXXXXXXXXXXX:role/StateMachineRole
      Events:
        CWSchedule:
          Type: Schedule
          Properties:
            Schedule: rate(1 minute)
            Name: ScheduleOne
            Description: Uno
            Enabled: false
```
This behavior is limiting where a Role is required to be re-used or places where there are constraints on Role naming conventions / path. 

### Proposal

The Resource should have a property for accepting the CloudWatch Schedule Role. If the field is left blank, a default one would be created. 

Things to consider:
[1] The [SAM documentation - Schedule](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-property-statemachine-statemachineschedule.html) will need to be updated

**Comment by jfuss** [comment]
Seems like a good addition. Updating labels.

**Comment by sidhujus** [comment]
feature added in [#3363](https://github.com/aws/serverless-application-model/pull/3363) and will be rolling out soon
